src/main/webapp/app/exercises/text/manage/tutor-effort/tutor-effort-statistics.component.ts

```diff
@@ -0,0 +1,135 @@
+                        labelString: 'Tutors',
+                    },
+                    ticks: {
+                        beginAtZero: true,
+                        stepSize: 1,
+                    },
+                },
+            ],
+        },
+    };
+
+    constructor(private textExerciseService: TextExerciseService, private textAssessmentService: TextAssessmentService, private route: ActivatedRoute) {}
+
+    ngOnInit(): void {
+        this.route.params.subscribe((params) => {
+            this.currentExerciseId = Number(params['exerciseId']);
+            this.currentCourseId = Number(params['courseId']);
+        });
+        this.loadTutorEfforts();
+    }
+
+    loadTutorEfforts() {
+        this.textExerciseService.calculateTutorEffort(this.currentExerciseId, this.currentCourseId).subscribe((tutorEffortResponse: TutorEffort[]) => {
+            this.handleTutorEffortResponse(tutorEffortResponse);
+        });
+        this.loadNumberOfTutorsInvolved();
+    }
+
+    /**
+     * Handler function to handle input data coming from service call.
+     * Separation enables better testing
+     * @param tutorEffortData - data to handle
+     */
+    handleTutorEffortResponse(tutorEffortData: TutorEffort[]) {
+        this.tutorEfforts = tutorEffortData;
+        if (!this.tutorEfforts) {
+            return;
+        }
+        this.numberOfSubmissions = this.tutorEfforts.reduce((n, { numberOfSubmissionsAssessed }) => n + numberOfSubmissionsAssessed, 0);
+        this.totalTimeSpent = this.tutorEfforts.reduce((n, { totalTimeSpentMinutes }) => n + totalTimeSpentMinutes, 0);
+        const avgTemp = this.totalTimeSpent === 0 ? 0 : this.numberOfSubmissions / this.totalTimeSpent;
+        this.averageTimeSpent = avgTemp ? Math.round((avgTemp + Number.EPSILON) * 100) / 100 : 0;
+        this.distributeEffortToSets();
+        this.chartDataSets[0].data = this.effortDistribution;
+    }
+
+    loadNumberOfTutorsInvolved() {
+        this.textAssessmentService.getNumberOfTutorsInvolvedInAssessment(this.currentCourseId, this.currentExerciseId).subscribe((response: number) => {
+            this.numberOfTutorsInvolvedInCourse = response;
+        });
+    }
+
+    /**
+     * Tutor Effort is distributed among the effortDistribution entries with each entry representing
+     * a corresponding index in the chartLables field.
+     * chartLabels.["0-10"] - corresponds to effortDistribution[0]
+     * chartLabels.["10-20"] - corresponds to effortDistribution[1]
+     * and so on. chartlabels is divided in steps of length 10, which is why division/10 and floor function is used.
+     */
+    distributeEffortToSets() {
+        this.effortDistribution = new Array<number>(this.chartLabels.length).fill(0);
+        this.tutorEfforts.forEach((effort) => {
+            const BUCKET_COUNT = this.chartLabels.length;
+            const time = Math.min(Math.floor(effort.totalTimeSpentMinutes / 10), BUCKET_COUNT - 1);
```

Comment:
without comment it's hard to understand what is happening here. 10 looks like a magic number, I would introduce a constant `BUCKET_TIME_SPAN` or something similar instead. Also the variable name `time` is not intuitive IMO.